pulsar-broker/src/test/java/org/apache/pulsar/client/api/DeadLetterTopicTest.java

```diff
@@ -154,28 +170,42 @@ public void testDeadLetterTopic() throws Exception {
 
         Producer<byte[]> producer = pulsarClient.newProducer(Schema.BYTES)
                 .topic(topic)
+                .enableChunking(produceLargeMessages)
+                .enableBatching(!produceLargeMessages)
                 .create();
 
+        Map<Integer, String> messageContent = new HashMap<>();
+
         for (int i = 0; i < sendMessages; i++) {
-            producer.send(String.format("Hello Pulsar [%d]", i).getBytes());
+            String data;
+            if (!produceLargeMessages) {
+                data = String.format("Hello Pulsar [%d]", i);
+            } else {
+                data = createMessagePayload(1024 * 10);
+            }
+            producer.newMessage().key(String.valueOf(i)).value(data.getBytes()).send();
+            messageContent.put(i, data);
         }
 
         producer.close();
 
         int totalReceived = 0;
         do {
             Message<byte[]> message = consumer.receive();
-            log.info("consumer received message : {} {}", message.getMessageId(), new String(message.getData()));
+            log.info("consumer received message : {}", message.getMessageId());
             totalReceived++;
         } while (totalReceived < sendMessages * (maxRedeliveryCount + 1));
 
         int totalInDeadLetter = 0;
         do {
             Message message = deadLetterConsumer.receive();
```

Comment:
This method does not add a timeout, some unexpected things will cause the CI not to finish.

Could we change `deadLetterConsumer.receive()` to `deadLetterConsumer.receive(timeout)`, or create a new thread to receive messages?